lib/src/validator/pubspec_typo.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../dice_coefficient.dart';
+import '../entrypoint.dart';
+import '../validator.dart';
+
+/// Validates that a package's pubspec does not contain any typos in its keys.
+class PubspecTypoValidator extends Validator {
+  PubspecTypoValidator(Entrypoint entrypoint) : super(entrypoint);
+
+  @override
+  Future validate() async {
+    final fields = entrypoint.root.pubspec.fields;
+
+    if (fields == null) return;
+
+    for (final key in fields.keys) {
+      var bestDiceCoefficient = 0.0;
+      var closestKey = '';
+
+      for (final validKey in _validPubspecKeys) {
+        final dice = diceCoefficient(key, validKey);
+        if (dice > bestDiceCoefficient) {
+          bestDiceCoefficient = dice;
+          closestKey = validKey;
+        }
+      }
+
+      // 0.73 is a magic value determined by looking at the most common typos
+      // in all the pubspecs on pub.dev.
+      if (bestDiceCoefficient >= 0.73 && bestDiceCoefficient < 1.0) {
+        warnings.add('$key is not a key recognizable by pub - '
+            'did you mean $closestKey?');
+      }
+
+      if (closestKey == 'author' || closestKey == 'authors') {
+        warnings.add('The "$closestKey" key is deprecated - Use a verified '
+            'publisher (https://dart.dev/tools/pub/verified-publishers) '
+            'instead.');
+      }
```

Comment:
Don't we have another warns about `author` and `authors`?

Reply:
Yes but not avthors